rfc/rfc023-x509credential.md

```diff
@@ -0,0 +1,453 @@
+certificate chain of the certificate. The verifier can then trust the information in the certificate by trusting the CA
+that signed the certificate.
+
+The chain of certificates can be viewed as a hierarchy, where the root certificate is the certificate is trusted, and
+signing is delegated to intermediate certificates. The root certificate is the certificate that is trusted by the
+holder. The holder maintains a list of trusted CAs that the holder trusts. The holder can then
+verify the signature of the certificate chain by verifying the signature of the CA that signed the intermediate
+certificate and the intermediate certificates that lead to the signing certificate.
+
+```asciidoc
+┌────────────────────┐
+│        CA          │
+└─────────┬──────────┘
+          │           
+┌─────────▼──────────┐
+│    Intermediate    │
+└─────────┬──────────┘
+          │           
+┌─────────▼──────────┐
+│    Intermediate    │
+└─────────┬──────────┘
+          │           
+┌─────────▼──────────┐
+│Signing Certificate │
+└────────────────────┘
+```
+
+### Using x509 for signing JWEs
+
+The JWE is a standard that is used to sign and encrypt JSON objects. Thus standard allows for the signing and encryption
+of JSON objects with certificates part of the a certificate chain. This allows for the signing of JSON objects with the
+private key of the certificate and the verification of the signature with the public key of the certificate, and the
+verification of the certificate chain with the public key of the CA. This is done by using the following headers fields:
+
+* x5c, the certificate chain as a list of base64 encoded certificates in the DER format, with the signing certificate
+  first and the root certificate last.
+* x5t, the thumbprint of the signing certificate.
+* x5t#S256, the thumbprint of the signing certificate as a SHA256 hash.
+
+### The `did:x509` DID Method
+
+The `did:x509` DID method is a method that can be used to create a Decentralized Identifier (DID) based on an x509
+certificate chain. This is done by creating a DID that is based on the root CA of the certificate chain. The did:x509
+method is used to specify specific attributes of the signing certificate to specify the holder of the signing
+certificate. By doing this, a did:x509 DID can be used to identify the holder of the signing certificate by specifica
+attributes that are assigned to the signing certificate. So, for example following did:x509:
+
+```
+did:x509:0:sha256:WE4P5dd8DnLHSkyHaIjhp4udlkF9LqoKwCvu9gl38jk::subject:C:US:ST:California:O:My%20Organisation
+```
+
+ties down the holder of the signing certificate by, first having a digitally signed certificate by the root CA with the
+thumbprint `WE4P5dd8DnLHSkyHaIjhp4udlkF9LqoKwCvu9gl38jk` and then having the following attributes in the certificate:
+
+* Subjects:
+  * C: US
+  * ST: California
+  * O: My Organisation
+
+The did:x509 defines various attribute types that can be used as attributes, such as:
```

Comment:
if this is the complete list, no need to specify "such as", since this is all there is